plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -116,24 +122,38 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
             is FirValueParameterSymbol -> VariableEmbedding(symbol.callableId.embedName(), type).toLocalVar()
 
             is FirPropertySymbol -> {
+
                 val varEmbedding = VariableEmbedding(symbol.callableId.embedName(), type)
+
                 if (symbol.isLocal) {
                     return varEmbedding.toLocalVar()
+                }
+
+                // When the symbol is not local, it may be global or belonging to a class
+                val receiver = propertyAccessExpression.dispatchReceiver.accept(this, data)
+
+                val anon = data.newAnonVar(varEmbedding.type)
+                data.addDeclaration(anon.toLocalVarDecl())
+
+                // Are we accessing a class field backed by a concrete field?
+                if (!symbol.hasBackingField) {
+                    val getter: FirPropertyAccessor = symbol.getterSymbol!!.fir
+                    val method = data.add(getter.symbol, getter.body)
+                    val methodCall = Stmt.MethodCall(method.name.mangled, listOf(receiver), listOf(anon.toLocalVar()))
+                    data.addStatement(methodCall)
                 } else {
-                    val receiver = propertyAccessExpression.dispatchReceiver.accept(this, data)
+
                     val fieldAccess = Exp.FieldAccess(receiver, varEmbedding.toField())
-                    val accPred = AccessPredicate.FieldAccessPredicate(fieldAccess, PermExp.FullPerm())
-                    val anon = data.newAnonVar(varEmbedding.type)
 
-                    data.addDeclaration(anon.toLocalVarDecl())
-                    // Inhale permission for the field before reading it.
-                    data.addStatement(Stmt.Inhale(accPred))
                     // Access the field and assign the value to a newly created anonymous variable.
-                    data.addStatement(Stmt.assign(anon.toLocalVar(), fieldAccess))
-                    // Exhale permission for the field and return the anonymous variable value
-                    data.addStatement(Stmt.Exhale(accPred))
-                    return anon.toLocalVar()
+                    inhaleAndExhaleStatement(
+                        AccessPredicate.FieldAccessPredicate(fieldAccess, PermExp.FullPerm()),
+                        Stmt.assign(anon.toLocalVar(), fieldAccess),
+                        data
+                    )
```

Comment:
This can be handled nicely with the new `StmtWithResultConversionContext`. Have a look at this function in the current `formal-verification` branch.